docs/learn.md

```diff
@@ -1,3 +1,153 @@
+in many ways: it can generate more readable and lighter code. It also generates
+one `.js` file per module, which makes it easier to integrate with existing
+JavaScript codebases. BuckleScript puts a big emphasis on communicating with
+JavaScript code through a rich collection of attributes applied to `external`
+primitives.
+
+#### Reason
+
+Around the same year, a project called [Reason](https://reasonml.github.io/)
+appears at Facebook. Led by Jordan Walke, the idea is to create an alternate
+syntax for OCaml that is closer to C and JavaScript. Even if Reason has no take
+on which platform the code is deployed —native applications binaries, or web
+applications using JavaScript as a target language—, BuckleScript adds first
+class support for Reason from early on. At that point, it becomes evident that
+the combination of Reason with BuckleScript is a great match: BuckleScript
+provides tools and infrastructure to work with JavaScript ecosystem, while
+Reason allows developers to write their programs in a syntax they are familiar
+with.
+
+Over time, and with help of other Facebook employees and the community providing
+bindings to pervasive JavaScript libraries like React.js with
+[`reason-react`](https://github.com/reasonml/reason-react/), the combination of
+Reason and BuckleScript gains adoption.
+
+#### BuckleScript gets rebranded
+
+However, at some point the goals of both BuckleScript and Reason projects become
+harder to reconcile. In August 2020, the BuckleScript team decides to rename to
+ReScript, stops adding support for the latest versions of the Reason parser, and
+replaces it with a new parser that changes the syntax. The reasons for the
+rebranding are explained in [the official ReScript blog
+post](https://rescript-lang.org/blog/bucklescript-is-rebranding).
+
+The rebranding is trying to ease onboarding and adoption of the ReScript
+language, giving the project more chances to compete with mainstream
+compiled-to-JavaScript languages like TypeScript. However, for many existing
+users of BuckleScript and Reason, it is the explicit confirmation of something
+that had been hinted implicitly before: ReScript goals are not compatible with
+providing a good integration with the OCaml ecosystem.
+
+#### Melange: back to OCaml
+
+This is where Melange comes in. A few weeks after the rebranding of BuckleScript
+to ReScript, António Monteiro starts working on a fork of BuckleScript with a
+simple (not easy) goal: replace the [Ninja build
+system](https://ninja-build.org/), which BuckleScript had been using from its
+creation, with [Dune](https://dune.build/), which is the most used build system
+for OCaml projects.
+
+This fork of BuckleScript is later named Melange. After finishing the switch
+from Ninja to Dune, several additional features were added to bring it closer to
+OCaml. Some examples are the upgrade of the compiler version used by Melange to
+4.12, or modelling the changes to the OCaml compiler that Melange uses as just a
```

Comment:
4.12 and, later, to 4.14.

Reply:
It seems I changed this at some point after the review, now it says:

> Some examples are the upgrade of the OCaml compiler version used by Melange, or modeling the changes...